main.c

```diff
@@ -132,7 +132,7 @@ static void reset_devices(void) {
     #endif
 }
 
-STATIC void start_mp(supervisor_allocation *heap) {
+STATIC void start_mp(supervisor_allocation *heap, supervisor_allocation *pystack, int pystack_size) {
```

Comment:
Although passing the stack allocation as a parameter is a sensible thing to do, the trouble here is that that parameter is dependent on MICROPY_ENABLE_PYSTACK. I don't think it is neat to #ifdef the function signature. So I would probably move the heap and stack  allocation calls inside `start_mp` to minimize #ifdefs and duplicated code.

Reply:
In stackless, pystack is being passed uninitilized. The warning is suppressed with a `#pragma`.
The function decleration is not `#ifdef`'ed.
Also, start_mp is being run multiple times. Meanwhile the way I am doing it, the pystack is allocated once.

Reply:
Perhaps it would be good to try to have it in `start_mp` regardless, so that it gets resized on reload.

Reply:
I do think you want to pass in the pystack allocation because you'll want to free it after mp completes. You don't need to pass in the size though because you can do `get_allocation_length()` like is done for the haep.

Reply:
The `mp_memory_t` struct handles storage and `get_allocation_length()` is used for both pystack and heap now.